File: platform/backend/app/schemas/template.py
"""Pydantic schemas para templates do marketplace."""

from datetime import datetime

from pydantic import BaseModel, ConfigDict


class TemplateResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: str
    slug: str
    name: str
    tagline: str
    description: str
    category: str
    tags: list[str]
    icon: str
    icon_bg: str
    version: str
    author: str
    deploy_count: int
    duration_estimate: str
    architecture_bullets: list[str]
    env_schema: list[dict]
    changelog: list[dict]
    published: bool
    created_at: datetime
    updated_at: datetime

File: platform/frontend/app/types/template.ts
export type TemplateCategory = "ETL" | "CRM" | "E-commerce" | "Analytics" | "Observability"

export interface EnvVarSchema {
  key: string
  label: string
  type: "text" | "password" | "url" | "cron" | "select"
  required: boolean
  placeholder?: string
  helper?: string
  default?: string
  options?: string[]
}

export interface TemplateChangelog {
  version: string
  date: string
  changes: string[]
}

export interface Template {
  slug: string
  name: string
  tagline: string
  description: string
  category: TemplateCategory
  tags: string[]
  icon: string
  iconBg: string
  version: string
  author: string
  deployCount: number
  durationEstimate: string
  architectureBullets: string[]
  envSchema: EnvVarSchema[]
  changelog: TemplateChangelog[]
  published: boolean
}
